Kernel/VM/Space.cpp

```diff
@@ -158,6 +158,63 @@ Region* Space::find_region_containing(const Range& range)
     return nullptr;
 }
 
+Vector<Region*> Space::find_regions_from_range(Range* range)
+{
+    Vector<Region*> regions = {};
+
+    ScopedSpinLock lock(m_lock);
+
+    // FIXME: Maybe take the cache from the single lookup?
+
+    size_t size = page_round_up(range->size());
+
+    bool found_one = true;
+    while (range && found_one) {
+        found_one = false;
+
+        for (auto& region : m_regions) {
+            if (region.vaddr() == range->base())
+                if (region.size() == size) {
+                    regions.append(&region);
+
+                    auto split_range = range->carve(region.range());
+                    range = split_range.is_empty() ? &(split_range.last()) : nullptr;
```

Comment:
I think that piece of code was for a exit condition to tell the while loop to breakout, because we used up all of the range to unmap, but this pice of code isnt used anymore